File: c++/arrays/iterating_over_arrays_and_reading_elements.cpp
#include <iostream>

int main()
{
  const char *array[5] = {"apple", "orange", "banana", "pear", "peach"};

  for (int i = 0; i < 3; i++)
    std::cout << "array[" << i << "]: " << array[i] << std::endl;

  std::cout << std::endl;

  for (int i = 0; i < 3; i++)
    std::cout << "*(array + " << i << "): " << *(array + i) << std::endl;

  std::cout << std::endl;

  const char **element_pointer = array;

  for (int i = 0; i < 3; i++)
    std::cout << "*(element_pointer++): " << *(element_pointer++) << std::endl;

  element_pointer = 0;

  std::cout << std::endl;
}

File: pascal/arrays/iterating_over_arrays_and_reading_elements.pas
(*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 *)

program IteratingOverArraysAndReadingElements;

var
  some_array: array[0..4] of string = ('apple', 'orange', 'banana', 'pear', 'peach');
  i: integer;

begin
  for i := 0 to high(some_array) do
    writeln('array[', i, ']: ', some_array[i]);

  writeln();
end.
